core/modules/layout/css/layout.admin.css

```diff
@@ -134,25 +134,24 @@
   width: 100%;
   min-height: 70px;
   margin: 0 0 15px;
-  padding: 8px;
+  padding: 8px 8px 16px 8px;
   border: 1px dashed #CCC;
   background: white;
   -webkit-box-sizing: border-box;
   box-sizing: border-box;
 }
 .layout-editor-region-title .label {
   float: left;
-  margin: 0 12px 0 0; /* LTR */
-  padding: 4px 0;
+  margin: 0;
+  padding: 4px 12px 4px 4px;
   line-height: 1.2;
+  text-transform: uppercase;
+  font-weight: normal;
+  color: #908990;
 }
 .layout-editor-region-content {
   min-height: 40px;
 }
-[dir="rtl"] .layout-editor-region-title .label {
-  margin-right: 0;
-  margin-left: 12px;
-}
```

Comment:
This needs to be changed to `padding-*` rules (and a 4px right margin) to support RTL styles.